### PR TITLE
Add platforms and roots to acme provisioner

### PR DESCRIPTION
**Pull request by maraino**
### Description

This PR adds platforms and roots to the acme provisioner.

### CODE REVIEW

File: provisioners.proto [hunk 1]
```diff
@@ -139,9 +139,17 @@ message ACMEProvisioner {
 		TLS_ALPN_O1 = 3;
 		DEVICE_ATTEST_01 = 4;
 	}
+	enum PlatformType {
+		NONE = 0;
```
Comment (hslatman) [inline provisioners.proto:143]: Would this be a meaningful "platform", or is `Unknown` better here? If the use case is that none of the other types are used, then `None` seems OK.

Comment (maraino) [reply provisioners.proto:143]: `UNKNOWN` cannot be used because it collides with ChallengeType's UNKNOWN.

Comment (hslatman) [reply provisioners.proto:143]: Bummer. Well, let's go with this then for now. We'll probably have to iterate on it a bit anyway

**Review by hslatman — APPROVED** [review]

File: provisioners.proto (hunk 1 above)
Comment (maraino) [reply provisioners.proto:143]: I won't tag it for now. Just use @main to use it.

**Pull request merged**